### PR TITLE
bugfix: inline cpi variables to fix stack violation warnings

### PR DESCRIPTION
**Pull request by sunguru98**
The token extensions macro is responsible for initializing extensions while creating a mint was having way too many repetitive stack variables that resulted in a "Stack size overset" warning while building programs. It wasn't occurring until there were way too many extensions initialized. Hence this fix inlines the variables directly into arguments to fix the warning that arose while building the WNS program.

### CODE REVIEW

**Review by kespinola — APPROVED** [review]
This change reduces the limit enough to be used in WNS but adding more expansion arms or more extensions in an instruction will likely overflow the stack. We will continue to monitor this within WNS and improve as we encounter issues.

**Comment by luke-truitt** [comment]
This issue applies generally to other Token Extensions projects as well, this resolves outstanding issues on the RWA Token Extension project as well. would be great to have in the next release.

**Review by acheroncrypto — CHANGES_REQUESTED** [review]
The constraint changes are OK but please don't modify versions of the crates.

**Comment by sunguru98** [comment]
My bad ser. Apologies.

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ The minor version will be incremented upon a breaking change and the patch versi
 
 ### Breaking
 
+## [0.30.1] - 2024-04-17
+
+### Fixes
+
+- syn: Eliminate variable allocations that build up stack space for token extension code generation ([#2913](https://github.com/coral-xyz/anchor/pull/2913)).
+
```
Comment (acheroncrypto) [inline CHANGELOG.md:24]: Could you add this change under the [Unreleased](https://github.com/coral-xyz/anchor/blob/8e8df76eda965ca6815dbdcb27f96c3dea018b3f/CHANGELOG.md#unreleased) section, and remove the new version section?

File: Cargo.lock
Comment (acheroncrypto) [inline Cargo.lock:1]: Could you revert this lock file change?

File: docs/src/pages/release-notes/changelog.md
Comment (acheroncrypto) [inline docs/src/pages/release-notes/changelog.md:1]: We only update release notes/changelog docs when we have a new release. Could you revert this change?

**Review by acheroncrypto — CHANGES_REQUESTED** [review]

**Comment by sunguru98** [comment]
Done ser @acheroncrypto

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ The minor version will be incremented upon a breaking change and the patch versi
 
 ### Fixes
 
-### Breaking
```
Comment (acheroncrypto) [inline CHANGELOG.md:17]: We should keep the "Breaking" section.

File: docs/src/pages/release-notes/0.30.0.md [hunk 3]
```diff
@@ -68,7 +68,7 @@ declare_program!(program_name);
 
 `program_name` is based on the file name of the IDL in `idls` directory, e.g. `idls/program_name.json` is required to exist in order for the above example to work.
 
-This works for both on-chain (CPI) and off-chain (RPC) usage, allowing program interactions without creating a [dependency hell](https://en.wikipedia.org/wiki/Dependency_hell). Check out [this](https://github.com/coral-xyz/anchor/blob/0.30.0/tests/declare-program/programs/declare-program/src/lib.rs) example for on-chain CPI usage.
+This works for both on-chain (CPI) and off-chain (RPC) usage, allowing program interactions without creating a [dependency hell](https://en.wikipedia.org/wiki/Dependency_hell). Check out [this](https://github.com/coral-xyz/anchor/blob/v0.30.0/tests/declare-program/programs/declare-program/src/lib.rs) example for on-chain CPI usage.
```
Comment (acheroncrypto) [inline docs/src/pages/release-notes/0.30.0.md:71]: This doesn't matter much, but it doesn't belong in this PR, and also it's a duplicate. You should get the changes from the `master` branch so that this doesn't show up as a new change.

**Review by acheroncrypto — CHANGES_REQUESTED** [review]
I'd like to avoid requesting changes, but this still has small issues.

**Comment by sunguru98** [comment]
Just wanting to make sure if thats all the changes you need? The feeling of not asking you for changes again is mutual here as well hehe. Hopefully this is the last one?

File: CHANGELOG.md [hunk 4]
```diff
@@ -16,6 +16,8 @@ The minor version will be incremented upon a breaking change and the patch versi
 
 ### Fixes
 
+- syn: Eliminate variable allocations that build up stack space for token extension code generation ([#2913](https://github.com/coral-xyz/anchor/pull/2913)).
```
Comment (acheroncrypto) [inline CHANGELOG.md:19]:
```suggestion
- lang: Eliminate variable allocations that build up stack space for token extension code generation ([#2913](https://github.com/coral-xyz/anchor/pull/2913)).
```

**Review by acheroncrypto — APPROVED** [review]
Thank you!

File: CHANGELOG.md (hunk 4 above)
Comment (acheroncrypto) [reply CHANGELOG.md:19]: Nevermind. I was going to commit this before merging it but can't because the branch is on an organization fork.
